lib/app.dart

```diff
@@ -23,7 +31,45 @@ class App {
     _showContent(_currentPage);
     _renderProgress();
     container.classes.toggle("hidden", false);
-    interactive.Interactive(interactiveContainer);
+
+    initFirebase();
+  }
+
+  void initFirebase() async {
+    await fb.init();
+    fb.firebaseAuth.onAuthStateChanged.listen(_fbAuthChanged);
+    loginButton.onClick.listen((_) => fb.googleLogin());
+    logoutButton.onClick.listen((_) => fb.signout());
+  }
+
+  void _fbAuthChanged(firebase.User user) {
+    if (user == null) {
+      print("User signed out");
+      loginModal.removeAttribute("hidden");
+      _interactiveInstance?.clear();
+      return;
+    }
+
+    if (!fb_constants.allowedEmailDomains
+        .any((domain) => user.email.endsWith(domain))) {
+      print("error email domain not allowed");
+      loginError
+        ..removeAttribute("hidden")
+        ..innerText = "Email domain not allowed";
+      return;
+    }
+
+    if (!user.emailVerified) {
```

Comment:
I don't think we need this - my guess is that email verification applies only when the user creates an account with an email and a password, and we would want to verify that the user has access to that email address, which doesn't really apply when using SSO/OAuth. I may be wrong as I haven't really used this property, could you have a look around to confirm? Thanks!

Reply:
Even with SSO, this may still apply - e.g. a new email created under the domain. 

I think this flag is an abstraction over the respective SSO's email verification - e.g. if the gmail account's email is verified or if the email you use for facebook is verified, if we are using facebook sso